### ISSUE TITLE
Handle UTF-8 BOM in utf8Decode

### ISSUE THREAD

**Issue by coreteam-co** [functional-streams-for-scala/fs2#1484]
I ran into some issues processing a CSV file which contains an UTF-8 BOM (Byte order mark, see https://en.wikipedia.org/wiki/Byte_order_mark). The Unicode Standard permits the BOM in UTF-8 but does not require or recommend its use. Unfortunately Microsoft uses it for example when saving an Excel file in CSV format.

I solved it with:

```
val stream: Stream[F, Byte] = ???
stream
   .through(fs2.text.utf8Decode)
   .map(s => if(s.startsWith("\ufeff")) s.substring(1) else s)
```

Would it make sense to make `fs2.text.utf8Decode` able to handle UTF-8 byte order mark? If so I can look into preparing a pull request.

**Comment by djspiewak** [comment]
Yes, this would most definitely make sense. I've been bitten by similar things as well.

**Comment by mpilquist** [comment]
A PR would be great!

**Comment by mpilquist** [comment]
@coreteam-co Are you working on a PR for this? If not, I can put one together but don't want to step on any work you have in progress. Thanks!

**Comment by coreteam-co** [comment]
@mpilquist Hi Michael - sorry for getting silent but I didn't have to get to it yet. Feel free to jump in.

**Comment by mpilquist** [comment]
Fixed by #1501
